sw/device/lib/uart.c

```diff
@@ -42,5 +42,16 @@ void uart_send_uint(uint32_t n, int bits) {
 }
 
 int uart_rcv_char(char *c) {
-  return dif_uart_byte_receive_polled(&uart0, (uint8_t *)c);
+  int ret;
+  size_t num_bytes = 0;
+  if (dif_uart_rx_bytes_available(&uart0, &num_bytes) == false) {
+    ret = -1;
```

Comment:
Here, just `return -1` and then don't have an `else` block, like so:

```c
if (dif_uart_rx_bytes_available(/*...*/)) {
  return -1;
}
if (num_bytes == 0) {
  return -1;
}

// ...
```

Reply:
Done.